seacrowd/sea_datasets/memolon/memolon.py

```diff
@@ -0,0 +1,140 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """\
+@inproceedings{buechel-etal-2020-learning-evaluating,
+    title = "Learning and Evaluating Emotion Lexicons for 91 Languages",
+    author = {Buechel, Sven  and
+      R{\"u}cker, Susanna  and
+      Hahn, Udo},
+    editor = "Jurafsky, Dan  and
+      Chai, Joyce  and
+      Schluter, Natalie  and
+      Tetreault, Joel",
+    booktitle = "Proceedings of the 58th Annual Meeting of the Association for Computational Linguistics",
+    month = jul,
+    year = "2020",
+    address = "Online",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/2020.acl-main.112",
+    doi = "10.18653/v1/2020.acl-main.112",
+    pages = "1202--1217",
+}
+"""
+
+_DATASETNAME = "memolon"
+
+_DESCRIPTION = """\
+MEmoLon is an emotion lexicons for 91 languages, each one covers eight emotional variables and comprises over 100k word entries. There are several versions of the lexicons, the difference being the choice of the expansion model.
+"""
+
+_HOMEPAGE = "https://zenodo.org/record/3756607/files/MTL_grouped.zip?download=1"
+
+_LICENSE = Licenses.MIT.value
+
+_URLS = {
+    _DATASETNAME: "https://zenodo.org/record/3756607/files/MTL_grouped.zip?download=1",
+}
+
+_SOURCE_VERSION = "1.0.0"
+
+_LANGUAGE_MAP = {"en": "English", "ceb": "Cebuano", "tl": "Tagalog", "id": "Indonesian", "su": "Sundanese", "jv": "Javanese", "ms": "Malay", "vi": "Vietnamese", "th": "Thai", "my": "Burmese", "zh": "Chinese"}
+
+_SUPPORTED_TASKS = [Tasks.EMOTION_CLASSIFICATION]
+
+
+def seacrowd_config_constructor(lang, schema, version):
```

Comment:
- let's put this inside the class?
- typing
```suggestion
def seacrowd_config_constructor(lang: str, schema: str, version: str) -> SEACrowdConfig:
```

Reply:
From what I observed, existing scripts defined `seacrowd_config_constructor` outside the class. I'm wondering if we have decided on a convention on whether to define it inside or outside the class? In my opinion, it would be better if there's such a convention. Please do let me know. 

Citing some scripts below (non-exhaustive list) that I found to have defined `seacrowd_config_constructor` outside the class:
https://github.com/SEACrowd/seacrowd-datahub/blob/16116789ff8e0bb5a3c87387586fcae1393af648/seacrowd/sea_datasets/nllb_seed/nllb_seed.py#L98

https://github.com/SEACrowd/seacrowd-datahub/blob/16116789ff8e0bb5a3c87387586fcae1393af648/seacrowd/sea_datasets/nusax_mt/nusax_mt.py#L57

Reply:
Since there's no response regarding conventions for `seacrowd_config_constructor` or similar, I'll keep it outside the class similar to the approved dataloaders I mentioned above.

For the typing, thanks for the suggestion. I already applied it and pushed the changes.